### PR TITLE
Create directory sync module

### PR DESCRIPTION
**Pull request by maxchehab**
This PR creates a Directory Sync module where one can:
- List Directories
  ```ts
  client.directorySync.listDirectories()
  ```
- List Users
  ```ts
  client.directorySync.listUsers("dir_edp_123")
  ```
- List Groups
  ```ts
  client.directorySync.listGroups("dir_edp_123")
  ```
- Get a User
  ```ts
  client.directorySync.getUser("dir_edp_123", "dir_usr_123")
  ```
- List User's Groups
  ```ts
  client.directorySync.listUsersGroups("dir_edp_123", "dir_usr_123")
  ```

### CODE REVIEW

**Comment by rjvani** [comment]
I think we need to add fetching a single user: https://docs.workos.com/directory-sync/api-reference#retrieve-a-single-user.

**Comment by maxchehab** [comment]
> I think we need to add fetching a single user: https://docs.workos.com/directory-sync/api-reference#retrieve-a-single-user.

Done

File: src/directory-sync/directory-sync.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    options?: PaginationOptions,
+  ): Promise<List<User>> {
+    const { data } = await this.workos.get(
+      `/directories/${directoryID}/users`,
+      options,
+    );
+    return data;
+  }
+
+  async listUsersGroups(directoryID: string, userID: string): Promise<Group[]> {
```
Comment (marktran) [inline src/directory-sync/directory-sync.ts:41]: Should this be `listUserGroups`?

We're grabbing groups for a single user, right?

Comment (maxchehab) [reply src/directory-sync/directory-sync.ts:41]: Shouldn't the noun be possessive?

Comment (marktran) [reply src/directory-sync/directory-sync.ts:41]: I always think about it as singular/plural. Don't think I've ever seen a possessive in API naming.

Comment (maxchehab) [reply src/directory-sync/directory-sync.ts:41]: 👍 https://github.com/workos-inc/workos-node/pull/126

**Pull request merged**